datafusion/physical-expr/src/aggregate/first_last.rs

```diff
@@ -417,7 +426,9 @@ impl Accumulator for LastValueAccumulator {
     fn state(&self) -> Result<Vec<ScalarValue>> {
         let mut result = vec![self.last.clone()];
         result.extend(self.orderings.clone());
-        result.push(ScalarValue::Boolean(Some(self.is_set)));
+        if !self.is_merge_called {
+            result.push(ScalarValue::Boolean(Some(self.is_set)));
+        }
```

Comment:
Why this is needed?

Reply:
Addressed here https://github.com/apache/arrow-datafusion/pull/7400#discussion_r1324934988